src/bp/ui-studio/src/web/views/OneFlow/sidePanel/Library/index.tsx

```diff
@@ -1,94 +1,180 @@
+  const [currentNamingEntity, setCurrentNamingEntity] = useState<NLU.EntityDefinition>(undefined)
+  const [forceUpdate, setForceUpdate] = useState(false)
+  const [entities, setEntities] = useState<NLU.EntityDefinition[]>([])
+
+  useEffect(() => {
+    async function fetchEntities() {
+      const res = await axios.get(`${window.BOT_API_PATH}/nlu/entities`)
+      setEntities(res.data)
+    }
+    // tslint:disable-next-line: no-floating-promises
+    fetchEntities()
+  }, [forceUpdate])
+
+  const createEntity = async (type: string) => {
+    const originalName = `${type}-entity`
+    let name = undefined
+    let index = 0
+    do {
+      name = `${originalName}${index ? `-${index}` : ''}`
+      index++
+    } while (entities.find(x => x.name === name))
+
+    const entity = {
+      id: name,
+      name: name,
+      type: type,
+      occurrences: []
+    }
+
+    await axios.post(`${window.BOT_API_PATH}/nlu/entities`, entity)
+    setForceUpdate(!forceUpdate)
   }
-}
 
-const Library: FC<Props> = props => {
-  const [isEditOpen, setEditOpen] = useState(false)
-  const [editItem, setEditItem] = useState('')
+  const deleteEntity = async (entity: NLU.EntityDefinition) => {
+    await axios.post(`${window.BOT_API_PATH}/nlu/entities/${entity.id}/delete`)
+    setForceUpdate(!forceUpdate)
+  }
+
+  const updateEntity = async (targetEntityId: string, entity: NLU.EntityDefinition) => {
+    await axios.post(`${window.BOT_API_PATH}/nlu/entities/${targetEntityId}`, entity)
+    const i = entities.findIndex(ent => ent.name === entity.name)
+    setEntities([...entities.slice(0, i), entity, ...entities.slice(i + 1)])
+  }
 
-  const getContextMenu = (element: LibraryElement) => {
+  const renderTable = (title: string, items: any[]) => {
     return (
-      <Menu>
-        <MenuItem
-          id="btn-remove"
-          icon="remove"
-          text={lang.tr('studio.flow.removeFromLibrary')}
-          onClick={() => {
-            props.removeElementFromLibrary(element.contentId)
-            props.refreshLibrary()
-          }}
-        />
-      </Menu>
+      <table
+        className={cx(
+          style.table,
+          'bp3-html-table bp3-html-table-striped bp3-html-table-bordered .bp3-html-table-condensed'
+        )}
+      >
+        <thead>
+          <tr>
+            <th>{title}</th>
+            <th></th>
+            <th></th>
+          </tr>
+        </thead>
+        <tbody>{renderTableRows(items)}</tbody>
+      </table>
+    )
+  }
+  const renderTableRows = (items: any[]) => {
+    return (
+      <Fragment>
+        {items &&
+          items.map((item, i) => (
+            <tr key={i}>
+              <td>
+                <Button text={item.label} onClick={item.click} />
+              </td>
+              <td>
+                <Button icon="edit" onClick={item.edit} />
+              </td>
+              <td>
+                <Button icon="delete" onClick={item.delete} />
+              </td>
```

Comment:
Mapping to add methods then mapping again to add feels very weird  This would be much better:
```js
<Button icon="delete" onClick={() =>  deleteEntity(item) } />
```

Reply:
This was done this way with the other menus in mind. Since there's going to be subworkflows and blocks in there also (see #3660)